lib/libc/common/source/time/asctime_r.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 Meta Platforms
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+
+#include <zephyr/sys/util.h>
+
+char *asctime_r(const struct tm *tp, char *buf)
+{
+	static const char wday_str[7][3] = {
+		"Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat",
+	};
+	static const char mon_str[12][3] = {
+		"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec",
+	};
+	char tmp[128];
+
+	if ((tp == NULL) || (buf == NULL) || (tp->tm_wday >= ARRAY_SIZE(wday_str)) ||
+	    (tp->tm_mon >= ARRAY_SIZE(mon_str))) {
+		return NULL;
+	}
+
+	snprintf(tmp, sizeof(tmp), "%.3s %.3s%3d %.2d:%.2d:%.2d %d\n", wday_str[tp->tm_wday],
```

Comment:
It's possible that the snprintf() parts could be broken up into smaller ones. I wonder if that would reduce the size of the buffer needed.

Alternatively, just print directly to buf. I would approve of that, and you would save a pile of stack space.